### ISSUE TITLE
Exported class can have method with the same name as imported

### ISSUE THREAD

**Issue by steida** [babel/babel#1160]
I updated to Babel 0.5, and getForm is somehow broken. Renaming to getForm as getFoo works. It's error in my code or in Babel?

```
import {getForm} from './store'

export default class Login extends React.Component {
  
  getForm() {
    return getForm().toJS()
  }

}
```

**Comment by tgriesser** [comment]
I believe I just hit this one as well - 

```js
import last from 'lodash/array/last'

export default class Container {

  last(key) {
    if (!this.has(key)) return
    return last(this.tokens.get(key))
  }

}
```

changing either of the `last` references will fix it

**Comment by sebmck** [comment]
Should be an easy fix, I'll look into it by the end of today and push out a
patch.

On Monday, 6 April 2015, Tim Griesser <notifications@github.com> wrote:

> I believe I just hit this one as well -
>
> import last from 'lodash/array/last'
>
> export default class Container {
>
>   last(key) {
>     if (!this.has(key)) return
>     return last(this.tokens.get(key))
>   }
>
> }
>
> changing either of the last references will fix it
>
> --
> Reply to this email directly or view it on GitHub
> <https://github.com/babel/babel/issues/1160#issuecomment-90099887>.
>


-- 
Sebastian McKenzie

**Comment by stefanpenner** [comment]
@sebmck thanks!

**Comment by jails** [comment]
Just stumbled on it too. But It has anything to do with `'export'` in my case and the following code:

```javascript
import events from 'events';

class Template {
  events() {
    return events;
  }
}

console.log(new Template().events());
```

produces the following error:  `Property 'events' of object #<Template> is not a function`

**Comment by sebmck** [comment]
I'lll push out a hacky fix now until I can work out a proper solution.

**Comment by sebmck** [comment]
Sorry for the delay, I've added all the examples posted in this issue as regression tests. Fixed as of 5.0.12, thanks :smile:

**Comment by jails** [comment]
Awesome work great :+1: !
